content-resources/src/main/java/flyway/oskari/V1_45_16__migrate_legacy_filtered_region_ids.java

```diff
@@ -0,0 +1,128 @@
+package flyway.oskari;
+
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.util.IOHelper;
+import fi.nls.oskari.util.JSONHelper;
+import org.flywaydb.core.api.migration.jdbc.JdbcMigration;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Migrates filtered regions (not supported yet) in bundle states legacy-block from sotkanet regions ids to features in regionsets.
+ */
+public class V1_45_16__migrate_legacy_filtered_region_ids implements JdbcMigration {
+
+    private static final Logger LOG = LogFactory.getLogger(V1_45_16__migrate_legacy_filtered_region_ids.class);
+
+    private Map<Integer, String> sotkaIdToRegionId = new HashMap<>();
+
+    class Bundle {
+        int view;
+        int bundle;
+        String state;
+    }
+
+    public void migrate(Connection conn) throws SQLException, IOException, JSONException {
+        populateRegionMapping();
+        List<Bundle> bundles = getStatesWithLegacyFilterRegions(conn);
+        for(Bundle bundle : bundles) {
+            if(migrateData(bundle)) {
+                updateData(conn, bundle);
+            }
+        }
+    }
+
+    private List<Bundle> getStatesWithLegacyFilterRegions(Connection conn) throws SQLException {
+        String sql = "SELECT view_id, bundle_id, state FROM portti_view_bundle_seq where bundle_id = (select id from portti_bundle where name = 'statsgrid') and state LIKE '%\"legacy\"%';";
+        List<Bundle> list = new ArrayList<>();
+        try (PreparedStatement ps = conn.prepareStatement(sql)) {
+            try (ResultSet rs = ps.executeQuery()) {
+                while(rs.next()) {
+                    Bundle ind = new Bundle();
+                    ind.view = rs.getInt("view_id");
+                    ind.bundle = rs.getInt("bundle_id");
+                    ind.state = rs.getString("state");
+                    list.add(ind);
+                }
+            }
+        }
+        return list;
+    }
+
+    private void populateRegionMapping() throws IOException, JSONException {
```

Comment:
Don't have to fix this, but this method could just return a Map<Integer, String>. Same thing in V1_45_14.